components/builder-depot/src/backend/s3.rs

```diff
@@ -103,6 +102,25 @@ impl S3Handler {
         }
     }
 
+    // This function checks whether an uploaded file
+    // exists in the configured s3 bucket. It should
+    // only get called from within an upload future.
+    fn object_exists(&self, object_key: &str) -> Result<()> {
+        let mut request = ListObjectsRequest::default();
+        request.bucket = self.bucket.clone();
+        request.prefix = Some(object_key.to_string());
+
+        match self.client.list_objects(&request).sync() {
+            Ok(object_list) => match object_list {
+                object_key => {
```

Comment:
I'm confused.  If I'm reading the docs correctly, `object_list` is of type `rusoto_s3::ListObjectsOutput`.  Then you match against that, comparing it to `object_key`, which is of type `&str`.  How does this ever succeed?

Reply:
It's not actually matching against the object list, we actually pass object key into the request function as an argument which forces the search to search for that specific file. This is more or less an unwrap so that we can log the value once we've resolved the initial map successfully.